### DISCUSSION TITLE
Why use mimetype over http.DetectContentType?

### DISCUSSION THREAD

**Discussion by johnmaguire** [gabriel-vasile/mimetype#356] [Q&A]
Hi there - I stumbled across this while looking at the awesome-go list. I'm planning to implement a file upload API in a project and was planning to use http.DetectContentType. I'm curious what the pros/cons are of using mimetype instead. Thanks!

**Comment by Fornax96** [comment]
http.DetectContentType only detects formats which are commonly served by web servers such as HTML, CSS, JS, image formats and audio and video content. If you have anything else like an ODT document it will use `application/octet-stream`, which is often too vague. Many clients rely on the mime type to be correct to open files with the right application.

One example which has bitten me before is that APK files can only be installed on Android when they are served with the mime type `application/vnd.android.package-archive`. If you serve it as anything else it will be opened as a zip file and it can't be installed.

DetectMimeType only detects 30 types, this library detects 172 types.
